source/common/common.c

```diff
@@ -22,6 +24,41 @@ int current_unix_timestamp(void) {
 	ularge.HighPart = file_time.dwHighDateTime;
 	return (long)((ularge.QuadPart - 116444736000000000) / 10000000L);
 }
+
+/*!
+ * @brief Sleep for the given number of seconds.
+ * @param seconds DWORD value representing the number of seconds to sleep.
+ * @remark This was implemented so that extended sleep times can be used (beyond the
+ *         49 day limit imposed by Sleep()). With this implementation it's possible
+ *         to wait for up to 3.1415926 metric buttloads of seconds.
+ *         NtDelayExecution takes a parameter that is measured in 100-ns units, and
+ *         negative values delay the thread relative to the current time. This is
+ *         a 63-bit value (1 bit for sign), which means it should be possible to wait
+ *         for approximately 29,227 years.
+ */

```

Comment:
If we're concerned about an undocumented function not being available, we would want the fallback case to loop anyway. So I'd say just loop, and then there is only one case to test.

Likely there's no real OS difference in how they're implemented anyway :) http://cvs.winehq.org/cvsweb/wine/scheduler/Attic/synchro.c?rev=1.48&content-type=text/x-cvsweb-markup